src/auth_v3.js

```diff
@@ -120,7 +120,25 @@ export function getClient (cozy, client) {
       token: client
     }
   })
-    .then((data) => new Client(data))
+    .then((data) => createClient(data, client))
+}
+
+// createClient returns a new Client instance given on object containing the
+// data of the client, from the API, and an old instance of the client.
+function createClient (data, oldClient) {
+  const newClient = new Client(data)
+  // we need to keep track of the registrationAccessToken since it is send
+  // only on registration. The GET /auth/register/:client-id endpoint does
+  // not return this token.
+  const shouldPassRegistration = (
+    !!oldClient &&
+    oldClient.registrationAccessToken !== '' &&
+    newClient.registrationAccessToken === ''
```

Comment:
With [this](https://github.com/cozy/cozy-client-js/blob/master/src/auth_v3.js#L14), we should always have a string in `registrationAccessToken`.